lib/bundler/definition.rb

```diff
@@ -953,8 +954,10 @@ def compute_requires
     def additional_base_requirements_for_resolve
       return [] unless @locked_gems && Bundler.feature_flag.only_update_to_newer_versions?
       @locked_gems.specs.reduce({}) do |requirements, locked_spec|
-        dep = Gem::Dependency.new(locked_spec.name, ">= #{locked_spec.version}")
-        requirements[locked_spec.name] = DepProxy.new(dep, locked_spec.platform)
+        name = locked_spec.name
+        next requirements if @locked_deps[name] != @dependencies_by_name[name]
```

Comment:
@segiddins I was just making a statement that `next` statements are one of my favourite because they mean less work == often more boot time perf. Was a stupid joke :)